### PR TITLE
OPSEXP-2055: address most ansible-lint issues

### PR DESCRIPTION
**Pull request by alxgomz**
did not address the following rules:
* role-name[path]: Needs more refactoring, tracked in https://alfresco.atlassian.net/browse/OPSEXP-2157
* galaxy: We don't push to Ansible galaxy at that point
* name[template]: I don't really agree with the rule - can be debated
* fcqn: Too much hits, will do in another PR

### CODE REVIEW

File: molecule/local/molecule.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ driver:
   name: docker
 platforms:
   - name: base-integration-instance
-    image: ${MOLECULE_ROLE_IMAGE:-rockylinux:8}
+    image: ${MOLECULE_ROLE_IMAGE:-rockylinux:8.7}
```
[the remark this answers is not in the document]
Comment (alxgomz) [reply molecule/local/molecule.yml:8]: True! Will deal with it in. https://alfresco.atlassian.net/browse/OPSEXP-2161

File: playbooks/secrets-init.yml [hunk 2]
```diff
@@ -11,12 +12,17 @@
...
         access_time: preserve
+        mode: "0600"
 
     - name: Peek into secrets.yml to discover if is an encrypted ansible-vault file
-      ansible.builtin.shell: head -1 {{ secrets_file }} | grep -q \$ANSIBLE_VAULT
+      ansible.builtin.shell:
+        executable: /bin/bash
+        cmd: |
+         set -o pipefail
+         head -1 {{ secrets_file }} | grep -q \$ANSIBLE_VAULT
```
[the remark this answers is not in the document]
Comment (alxgomz) [reply playbooks/secrets-init.yml:22]: This Ansible-lint rule: https://github.com/ansible/ansible-lint/blob/c2a935bcab2f8a2572d47e2d4ca521fe06243080/src/ansiblelint/rules/risky_shell_pipe.md

File: playbooks/secrets-init.yml [hunk 3]
```diff
@@ -48,7 +54,10 @@
 
         - name: Populate secrets.yml with missing secrets
           ansible.builtin.shell: "{{ base_folder }}/scripts/generate-secret.sh -s {{ item }} -m {{ gs_mode }} >> {{ secrets_file }}"
+          changed_when: true
```
[the remark this answers is not in the document]
Comment (alxgomz) [reply playbooks/secrets-init.yml:57]: yes it is!

File: playbooks/secrets-init.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gionn) [reply playbooks/secrets-init.yml:22]: good linter

**Review by gionn — APPROVED** [review]
overall LGTM

**Pull request merged**